### PR TITLE
Specific Configs to GethDebugTracerConfig + generic config build method for GethDebugTracingOptions

### PR DESCRIPTION
**Pull request by halo3mic**
## Motivation

1. Simpler conversion from specific tracer config to GethDebugTracerConfig
2. Single config build method on GethDebugTracingOptions that covers all configs 

## Solution

1. Impl From for CallConfig, PreStateConfig, MuxConfig
2. Introduce generic build method for trace configs on GethDebugTracingOptions

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

**Review by DaniPopes — APPROVED** [review]

**Comment by DaniPopes** [comment]
cc @mattsse

File: crates/rpc-types-trace/src/geth/mod.rs [hunk 1]
```diff
@@ -311,17 +329,12 @@ impl GethDebugTracingOptions {
...
-    pub fn prestate_config(mut self, config: PreStateConfig) -> Self {
-        self.tracer_config =
-            GethDebugTracerConfig(serde_json::to_value(config).expect("is serializable"));
+    /// Sets the tracer config
+    pub fn with_config<T>(mut self, config: T) -> Self
+    where
+        T: Into<GethDebugTracerConfig>,
+    {
+        self.tracer_config = config.into();
         self
```
Comment (mattsse) [inline crates/rpc-types-trace/src/geth/mod.rs:338]: can we keep this functions?

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/rpc-types-trace/src/geth/mod.rs (hunk 1 above)
Comment (mattsse) [reply crates/rpc-types-trace/src/geth/mod.rs:338]: yeah, and having them doesn't hurt

**Comment by yash-atreya** [comment]
@halo3mic are you still on this? Else I can take this over, we'd like to wrap it up

**Review by mattsse — APPROVED** [review]

**Pull request merged**